parsers/ocaml.c

```diff
@@ -1809,9 +2024,20 @@ static void localScope (vString * const ident, ocaToken what)
 static void computeModuleName ( void )
 {
 	/* in Ocaml the file name define a module.
-	 * so we define a module =)
+	 * so we define a module if the file has
+	 * things in it. =)
 	 */
 	const char *filename = getSourceFileName ();
+
+	FILE *file = fopen (filename, "r");
+	fseek (file, 0, SEEK_END);
+	long fileSize = ftell (file);
+	fclose (file);	// CLOSE!
+	if (fileSize == 0)
+	{
+		return;
+	}
```

Comment:
ctags can emit an ordered tag list, which is what emacs/semantic uses, so unfortunately I don't think that's viable. Token-presence checking looks like a good possibility, and I'm a little annoyed that I didn't consider that as C-style File I/O is really not my forte.